core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -209,6 +222,26 @@ object ConfigSource {
       valueDelimiter = valueDelimiter
     )
 
+  /**
+   *
+   * Loop through all command line arguments, looking for the form:
+   * --xyz=xyz
+   * --xyz xyz
+   * -xyz=xyz
+   * -xyz xyz
+   */
+  private[config] def argsAsKeyValues(args: List[String]): List[(String, String)] =
+    args.flatMap { s =>
+      if (s.startsWith("-") && s.contains("="))
+        s.substring(2).split('=').toList match {
+          case k :: v :: Nil => List(k, v)
+          case _             => Nil
+        }
+      else List(s)
```

Comment:
We expect some sort of smart constructors here.
We don't want to end up having key-value pairs for `-=hello` etc

Reply:
Or may be not. Let's see. It's good for now. But could you add an issue page to have a follow up PR on command-line args, to keep the users know we are working on it.